packages/salesforce/src/Utils.js

```diff
@@ -134,3 +134,69 @@ export async function pollJobResult(conn, job, pollInterval, pollTimeout) {
     attempt++;
   }
 }
+
+/**
+ * Flattens an array of nested objects to a maximum of two levels deep.
+ * - Each object in the array is flattened to have dot-separated keys.
+ * - Throws an error if the input is not an array or if a nested object exceeds two levels deep.
+ *
+ * @param {Array<Object>} input - An array of objects to flatten.
+ * @returns {Array<Object>} - An array of flattened objects.
+ * @throws {Error} - Throws an error if the input is not an array of objects or if nesting exceeds two levels.
+ */
+
+export function flattenData(input) {
+  if (!Array.isArray(input)) {
+    throw new Error('Input must be an array of objects.');
+  }
+
+  return input.map(item => flattenSingleObject(item));
+}
+
+const flattenSingleObject = (obj, parentKey = '', currentDepth = 0) => {
+  const result = {};
+  Object.entries(obj).forEach(([key, value]) => {
+    const newKey = parentKey ? `${parentKey}.${key}` : key;
+
+    if (typeof value === 'object' && value !== null && !Array.isArray(value)) {
+      if (currentDepth < 1) {
+        Object.assign(
+          result,
+          flattenSingleObject(value, newKey, currentDepth + 1)
+        );
+      } else {
+        throw new Error(
+          `Nested object with key ${newKey} exceeds the allowed depth`
+        );
+      }
+    } else {
+      result[newKey] = value;
+    }
+  });
+  return result;
+};
+
+/**
+ * Validates that no keys in the input object or array of objects contain dots ('.').
+ * Throws an error if a key containing a dot is found.
+ *
+ * @param {Object|Array<Object>} input - The object or array of objects to validate.
+ * @throws {Error} - Throws an error if a key containing a dot is found or if the input is invalid.
+ */
+export function validateNoDotKeys(input) {
```

Comment:
This is sort of thing (testing something and throwing) is usually known as an assertion - so I'd prefer `assertNoDotKeys` or even better `assertNoNesting`

Reply:
Changed to `assertNoNesting`